### ISSUE TITLE
Losing key with (key,value) form of ng-repeat

### ISSUE THREAD

**Issue by mcavanagh** [fmquaglia/ngOrderObjectBy#6]
With the following markup:
```
<tr ng-repeat="(teamName, teamData) in tableTeams | orderObjectBy: 'points'">
    <td>{{teamName}}</td>
    <td>{{teamData.points}}</td>
    <td>{{teamData.goalDifference}}</td>
</tr>
```

and model:

```
$scope.tableTeams = {
    "team1": {
        points: 34,
        goalDifference: 9
    },
    // etc...
}
```

the `{{teamName}}` value in the template shows a numeric index instead of the team name.  When using the standard angular `orderBy`, the name is preserved.  Is this something which can be easily fixed?

Thanks.

**Comment by fmquaglia** [comment]
@mcavanagh for the time being only dot notation is supported to access the ordered object's keys.
I would suggest as work around to have the name (and any metadata) as another attribute.

PRs are welcome. 😊

**Comment by carande** [comment]
I ran into this issue as well, and indeed solved it by appending the key as another attribute of the array element: 
```
        var filtered = [];
        angular.forEach(items, function (item, key) {
            item.objectID = key; // Need to keep info from object key
            filtered.push(item);
        });
```

**Comment by fmquaglia** [comment]
@carande I am curious about the context of your work around. Would you put something in place to see it live?

**Comment by carande** [comment]
Just getting back to this now... I'm not too familiar with how to set up a
"live" demo, but I sent a pull request showing how I edited your code.  Let
me know if that makes sense - this may not be the best way to do it, but it
was a solution that worked for me.

**Comment by jonalling** [comment]
I also would like a solution to this, although I am not skilled enough to contribute.

**Comment by fmquaglia** [comment]
This has been fixed by @carande in #9 and #16 
Thanks a lot buddy!

**Comment by carande** [comment]
Sure thing!
